integration/osio_provider_test.go

```diff
@@ -0,0 +1,106 @@
+package integration
```

Comment:
Follow below steps to run test.

**Steps:**
1. check out i1988 branch.
2. go generate
3. go build -o dist/traefik ./cmd/traefik/
4. go test -v ./integration/ -integration -osio

**Output:**
```
=== RUN   Test
Initialize OSIOAuth
resolved..
+
time="2018-04-06T17:17:48+05:30" level=info msg="req res.StatusCode=200" 
+
time="2018-04-06T17:17:51+05:30" level=info msg="req res.StatusCode=200" 
+
time="2018-04-06T17:17:54+05:30" level=info msg="req res.StatusCode=200" 
+
time="2018-04-06T17:17:57+05:30" level=info msg="req res.StatusCode=404" 
+
time="2018-04-06T17:18:00+05:30" level=info msg="req res.StatusCode=404" 
+
time="2018-04-06T17:18:03+05:30" level=info msg="req res.StatusCode=200" 
+
time="2018-04-06T17:18:06+05:30" level=info msg="req res.StatusCode=404" 
+
time="2018-04-06T17:18:09+05:30" level=info msg="req res.StatusCode=404" 
+
time="2018-04-06T17:18:12+05:30" level=info msg="req res.StatusCode=200" 
+
time="2018-04-06T17:18:15+05:30" level=info msg="req res.StatusCode=200" 
OK: 1 passed
--- PASS: Test (30.03s)
PASS
ok  	github.com/containous/traefik/integration	30.044s
```

**Check** - same request returning - sometimes 200 .. sometimes 404.